apps/console/src/features/authentication-flow-builder/components/script-editor-panel/script-based-flow-switch.tsx

```diff
@@ -0,0 +1,174 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Accordion from "@oxygen-ui/react/Accordion";
+import AccordionDetails from "@oxygen-ui/react/AccordionDetails";
+import AccordionSummary from "@oxygen-ui/react/AccordionSummary";
+import Box from "@oxygen-ui/react/Box";
+import Grid from "@oxygen-ui/react/Grid";
+import Switch from "@oxygen-ui/react/Switch";
+import Typography from "@oxygen-ui/react/Typography";
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import React, {
+    PropsWithChildren,
+    ReactElement,
+    useEffect,
+    useState
+} from "react";
+import { useTranslation } from "react-i18next";
+import AdaptiveScriptResetConfirmationModal from "./adaptive-script-reset-confirmation-modal";
+import ScriptEditorPanel from "./script-editor-panel";
+import { AdaptiveScriptUtils } from "../../../applications/utils/adaptive-script-utils";
+import useAuthenticationFlow from "../../hooks/use-authentication-flow";
+import "./script-based-flow-switch.scss";
+
+/**
+ * Proptypes for the Script Based Flow switching component.
+ */
+export interface ScriptBasedFlowSwitchPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * Make the form read only.
+     */
+    readOnly?: boolean;
+}
+
+/**
+ * Script based flow switch component.
+ *
+ * @param props - Props injected to the component.
+ * @returns Script based flow switch component.
+ */
+const ScriptBasedFlowSwitch = (props: PropsWithChildren<ScriptBasedFlowSwitchPropsInterface>): ReactElement => {
+    const { readOnly, "data-componentid": componentId } = props;
+
+    const { t } = useTranslation();
+
+    const {
+        authenticationSequence,
+        isConditionalAuthenticationEnabled,
+        onConditionalAuthenticationToggle,
+        updateAuthenticationSequence
+    } = useAuthenticationFlow();
+
+    const [ showScriptResetWarning, setShowScriptResetWarning ] = useState<boolean>(false);
+
+    /**
+     * This useEffect is responsible for deciding whether
+     * the adaptive script section should be enabled or not.
+     */
+    useEffect(() => {
+        if (authenticationSequence?.script
+                && !AdaptiveScriptUtils.isDefaultScript(
+                    authenticationSequence.script,
+                    authenticationSequence?.steps?.length
+                )
+                && !AdaptiveScriptUtils.isEmptyScript(authenticationSequence?.script)) {
+            onConditionalAuthenticationToggle(true);
+
+            return;
+        }
+
+        onConditionalAuthenticationToggle(false);
+    }, [ authenticationSequence.script ]);
+
+    /**
+     * This function will handle the on and off
+     * actions of the switch component.
+     */
```

Comment:
Do we need this comment as function name is self explanatory?

Reply:
Otherwise, it gives lint errors. Need to add doc comments for all functions.